### DISCUSSION TITLE
Parent child issues

### DISCUSSION THREAD

**Discussion by gabog** [community/community#7813] [Projects and Issues] [answered]
It would be nice if there was an easy way of creating parent child relationships between issues (within the same repo or across repos).

This would allow us to plan using an Epics->Feature->Story->Task hierarchy for tracking complex issues.

**Comment by 0Grit** [comment]
Being able to see this in the sheet view would be EXTREMELY helpful to my planning process.

https://github.com/orgs/golioth/projects/4/views/12

**Reply by bronzeson** [reply]
Having it as a field in the sheets view would be super nice!

Field: Parent Issue
Then when you create a new issue, you can search within that field from sheet view for an issue you're looking for to assign it a parent.

**Reply by zonofthor** [reply]
Yes yes yes! 

I spent a few days moving a large project from Clubhouse and after trying out various configurations finally found a pathway for Epics using exactly same approach. Came to exactly same conclusion, add a column with link to parent issue where current issue is referenced.

That´s all that´s needed and would immediately lend Epic support to the new Projects beta!

**Comment by white-sdev** [comment]
Please! make them with infinite sublevels, It is very anoying to find yourself limited by just 1 level, this will be deal braker for me :(
https://github.com/github/feedback/discussions/5714, https://github.com/github/feedback/discussions/9913

**Comment by Jotr38487** [comment]
it would be nice also if you could get more direct commands used across atleast ##most devices with java,c##,.json etc.

**Comment by rileybroughten** [comment, marked as answer]
Hello! 👋 Tasklists add support for issue hierarchy at GitHub and are currently in Private Beta! You can request access and join the waitlist [here](https://github.com/features/issues/signup).

For more information on how these work, check out the [documentation](https://docs.github.com/en/issues/tracking-your-work-with-issues/about-tasklists).

**Reply by Mohamed-Kassem-com** [reply]
Hi @rileybroughten 
Is nesting issues available in Tasklists?

So Tasklist would look like:
- Task 1
  - Sub task 1
  - Sub task 2
- Task 2
  - Sub task 1
  - Sub task 2

**Reply by rileybroughten** [reply]
Hello @Mohamed-Kassem-com, yes you can create multiple tasklists within a single issue to specify separate sub-tasks! Let me know if you have any other questions or feedback as you get started 👍

**Reply by cmeyer** [reply]
As @rileybroughten states, you can have multiple task lists per issue in separate lists, but I don't see any way to nest them.

How nesting would even work? For instance, if issue 1 has task list A which includes issue 2 with task list B and C, should the items in issue 2 tasks B and/or C automatically appear in the display for issue 1? And how deep would it go?

Maybe there could be a way to insert the task list from another issue explicitly with some notation like `#64^tasklist1`.

Also, if I indent a task, it still appears but at the left-most level in the list. There seems to be only a single level of indentation. Perhaps the display could honor the indentation in the markup for ad-hoc nesting?

**Reply by gajop-ptx** [reply]
In the context of GitHub Projects, I'm not sure it's appropriate to name this as the answer, as per the link, this is in private beta and onboarding has stopped.

> Tasklists are in private beta and subject to change. We have currently paused onboarding to the beta. Tasklists build upon the previous iteration of [task lists](https://docs.github.com/en/get-started/writing-on-github/working-with-advanced-formatting/about-task-lists).

I haven't had the chance to try it out, but judging by the screenshots I don't think it's sufficiently good when used outside of standalone repo issue trackers.

In the publicly released version of tasks lists, we can add multiple tasks in an issue, and the parent -> child issue link is visible in both the parent and child issues (great!). 

However, you lose all the goodies of GitHub Projects to the point that it's almost unrelated to GitHub Projects and is a different way of doing things. 
The big win of GitHub Projects are Tables, Kanban, Roadmap, with bulk editing, multiple views on the same data, etc. None of this is possible with task lists.
Further, even if you add those issues to the GitHub Project so you can manage it properly, you now have child issues showing along with parent issues, making it a UI mess (different levels of abstractions, unclear what is parent/child, etc)

Is there any reason not to add a simple (collapsable) parent-child nested systems in GitHub Projects? This seems like a simple & intuitive way of managing parent/child relationship that I've seen in different places (Notion I think uses it). 

```
↓ Parent1  | Column1 | Column2 |
  - Child1 | Column1 | Column2 |
  - Child2 | Column1 | Column2 |
- Parent2  | Column1 | Column2 |
- Parent3  | Column1 | Column2 |
```

**Reply by TechinMama** [reply]
@rileybroughten I've just noticed this comment. What does joining the waitlist do? Does this give the org access to Private Beta and function as some sort of UAT testing until the feature is rolled out to production?

**Reply by rileybroughten** [reply]
See the comment below in https://github.com/orgs/community/discussions/7813#discussioncomment-10682057 😁

**Comment by pingtobuild** [comment]
any update on this? when releasing to production?

**Comment by TechinMama** [comment]
I am toying around with projects and found this issue while searching for similar capability. It would be extremely helpful to be able to visually look at a parent issue and see the child issues in a list format.

**Comment by zofiasp** [comment]
Looking around for similar solution myself. I need to be able to have a custom field saying: Parent. And be able to choose a different issue as a parent. The parent would then have a component that would display all linked issues. That would be extremely helpful for linking "Epic" issuetypes with "user story" issuetypes and then tasks etc.

**Comment by jsilva-js** [comment]
none update?

**Comment by mitchezugwu** [comment]
Also hoping there is a resolution on its way to this request.  It would be a game changer to have some sort of parent child relationship functionality availble within Github projects for issues.

**Comment by data-if** [comment]
any updates?

**Comment by rileybroughten** [comment]
Hi all! We currently have a beta in progress for this, you can see more info in the below discussions for sub-issues and issue types: 

- #131957 
- #112806 

Access to these beta also gives you access to
- #132506 

If you would like to join this beta please let me know your organization names below and I will add you in. I will add a reaction to your comment once I have added you. Thanks and please share your feedback once you have tried it!

**Reply by nanisnice** [reply]
Hi @rileybroughten, could our org be included (https://github.com/nicevrs) We're looking forward to test these new features :)

**Reply by aitor** [reply]
Hi @rileybroughten! Could please add https://github.com/devengoapp to the beta? Thank you!

**Reply by alamiljoseph** [reply]
Hi @rileybroughten, we are also interested. Please add https://github.com/jxception-org. Thanks.

**Comment by rileybroughten** [comment]
👋 Check out the [changelog](https://github.blog/changelog/2024-10-01-evolving-github-issues-public-preview/) announcement where you can sign up to gain access to sub-issues, issue types, and advanced search if you do not have access already ✨
